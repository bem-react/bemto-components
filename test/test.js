const React = require('react');
const bemto = require('..');
const renderer = require('react-test-renderer');

const testSnapshot = function(tag, props, children) {
  const component = renderer.create(
    React.createElement(
      tag,
      props,
      children
    )
  );

  expect(component.toJSON()).toMatchSnapshot();
}

test('with tag name and external className', () => {
  const tag = bemto('div');
  testSnapshot(
    bemto('div'),
    { className: 'block', _mod: 'value' }
  );
});

test('with boolean modifier', () => {
  testSnapshot(
    bemto('div'),
    { className: 'block', _mod: true }
  );
});

test('with class name only', () => {
  testSnapshot(
    bemto('.block'),
    { _mod: 'value' }
  );
});

test('with tag and explicit class name', () => {
  testSnapshot(
    bemto('span.block'),
    { _mod: 'value' }
  );
});

test('with tag and explicit id and class name', () => {
  testSnapshot(
    bemto('span#id.block'),
    { _mod: 'value' }
  );
});

test('with multiple classes', () => {
  testSnapshot(
    bemto('.block1.block2'),
    { _mod: 'value' }
  );
});

test('with multiple external classes', () => {
  testSnapshot(
    bemto('div'),
    {
      className: 'block1 block2',
      _mod: 'value'
    }
  );
});
