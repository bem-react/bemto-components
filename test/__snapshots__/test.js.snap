// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with boolean modifier 1`] = `
<div
  className="block block_mod"
/>
`;

exports[`with class name only 1`] = `
<div
  className="block block_mod_value"
/>
`;

exports[`with multiple classes 1`] = `
<div
  className="block1 block2 block1_mod_value block2_mod_value"
/>
`;

exports[`with multiple external classes 1`] = `
<div
  className="block1 block2 block1_mod_value block2_mod_value"
/>
`;

exports[`with tag and explicit class name 1`] = `
<div
  className="block block_mod_value"
/>
`;

exports[`with tag and explicit id and class name 1`] = `
<div
  className="block block_mod_value"
  id="id"
/>
`;

exports[`with tag name and external className 1`] = `
<div
  className="block block_mod_value"
/>
`;
